packages/react-reconciler/src/ReactChildFiber.new.js

```diff
@@ -1221,21 +1221,21 @@ function ChildReconciler(shouldTrackSideEffects) {
     // not as a fragment. Nested arrays on the other hand will be treated as
     // fragment nodes. Recursion happens at the normal flow.
 
+    let isObject = typeof newChild === 'object' && newChild !== null;
+
     // Handle top level unkeyed fragments as if they were arrays.
     // This leads to an ambiguity between <>{[...]}</> and <>...</>.
     // We treat the ambiguous cases above the same.
     const isUnkeyedTopLevelFragment =
-      typeof newChild === 'object' &&
-      newChild !== null &&
+      isObject &&
       newChild.type === REACT_FRAGMENT_TYPE &&
       newChild.key === null;
     if (isUnkeyedTopLevelFragment) {
       newChild = newChild.props.children;
+      // Handle object types
+      isObject = typeof newChild === 'object' && newChild !== null;
```

Comment:
What is the profit of such a solution?

Reply:
> What is the profit of such a solution?

I'm sorry, I didn't explain the motive clearly.

`isObject = typeof newChild === 'object' && newChild !== nul`

motivation: I want to reduce the number of judgments of `isObject`.

Because most of the time,  `newChild.type` is not equal to `REACT_FRAGMENT_TYPE`, so when  `newChild.type` is not equal to `REACT_FRAGMENT_TYPE`, you don't have to compute `isObject` again.